src/include/alloc.h

```diff
@@ -36,6 +36,17 @@ static ncclResult_t ncclCalloc(T** ptr, size_t nelem) {
   return ncclSuccess;
 }
 
+template <typename T>
+static ncclResult_t ncclRealloc(T** ptr, size_t nelem) {
```

Comment:
NCCL has a structure for these sort of allocation, allocate a #define value and only use a subset if you need to. For example look at how scclAlgorithms is allocation in comm.

Reply:
I went for dynamic allocation since the new registrations are purely a host side feature.